File: ontrack-extension-stale/src/main/java/net/nemerosa/ontrack/extension/stale/AutoDisablingBranchPatternsMode.kt
package net.nemerosa.ontrack.extension.stale

/**
 * What should be done of branches matching a pattern?
 */
enum class AutoDisablingBranchPatternsMode {

    /**
     * Always keeping the matching branch
     */
    KEEP,

    /**
     * Always keeping the N last branches matching this pattern (based on semantic versioning)
     */
    KEEP_LAST,

    /**
     * Disabling the matching branch
     */
    DISABLE,

}

File: ontrack-web-core/components/extension/stale/AutoDisablingBranchPatternsMode.js
export const modes = [
    {
        value: 'KEEP',
        label: 'Keep',
    },
    {
        value: 'DISABLE',
        label: 'Disable ',
    },
    {
        value: 'KEEP_LAST',
        label: 'Keep last ',
    },
]

export default function AutoDisablingBranchPatternsMode({mode}) {
    const item = modes.find(it => it.value === mode)
    return item?.label
}